samples/Network/deny-public-ip-on-nic/azurepolicy.json

```diff
@@ -0,0 +1,42 @@
+{
+    "properties": {
+      "displayName": "Deny adding public IP Address to a virtual network interface",
+      "description": "Deny the ability to add a public IP Address to a virtual network interface.",
+      "policyType": "Custom",
+      "mode": "Indexed",
+      "metadata": {
+        "category": "Network"
+      },
+      "parameters": {
+        "effect": {
+          "type": "String",
+          "metadata": {
+            "displayName": "Effect",
+            "description": "Enable or disable the execution of the policy"
+          },
+          "allowedValues": [
+            "Disabled",
+            "Deny"
+          ],
+          "defaultValue": "Deny"
+        }
+      },
+      "policyRule": {
+        "if": {
+          "allOf": [
+            {
+              "field": "type",
+              "equals": "Microsoft.Network/networkInterfaces"
+            },
+            {
+              "field": "Microsoft.Network/networkInterfaces/ipconfigurations[*].publicIpAddress.id",
+              "exists": true
+            }
```

Comment:
This will only be true if the publicIpAddress exists on ALL of the ipConfigurations.
It will not block if there are several ipConfigurations and just one of them has a pip

Reply:
Just realized there's a built-in policy for this: "id": "/providers/Microsoft.Authorization/policyDefinitions/83a86a26-fd1f-447c-b59d-e51f44264114",

If you want, you can copy that logic here (doesn't hurt to have built-ins in the samples repo), or just abandon the PR. Sorry I didn't catch this earlier  - didn't mean to waste your time with minor comments.

---
In reply to: [312577762](https://github.com/Azure/azure-policy/pull/453#discussion_r312577762) [](ancestors = 312577762)